src/main/java/acme/entities/code_audits/CodeAudit.java

```diff
@@ -0,0 +1,49 @@
+
+package acme.entities.code_audits;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Past;
+import javax.validation.constraints.Pattern;
+
+import org.hibernate.validator.constraints.Length;
+
+import acme.client.data.AbstractEntity;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Setter
+@Getter
+public class CodeAudit extends AbstractEntity {
+
+	//Serialisation identifier -----------------------------------------------------------------
+
+	private static final long	serialVersionUID	= 1L;
+
+	//Attributes --------------------------------------------------------------------------------
+
+	@NotBlank
+	@Column(unique = true)
+	@Pattern(regexp = "[A-Z]{1,3}-[0-9]{3}")
+	String						code;
+
+	@Past
+	@NotNull
+	Date						executionDate;
+
+	@NotNull
+	CodeAuditType				type;
+
+	@NotBlank
+	@Length(max = 100)
+	String						correctiveAction;
+
```

Comment:
I don't know if "mark" property can be nullable. If it is, put the `@Nullable` annotation, otherwise, use the `@NotNull` annotation.